### ISSUE TITLE
brace-style "allman" fails with switch

### ISSUE THREAD

**Issue by aparajita** [eslint/eslint#4002]
**What version are you using?**

v1.5.1

**What did you do?**

Using `"brace-style": [2, "allman"]`, I wrote the following code:

```js
switch (foo)
{
    case 1:
        break;
}
```

**What happened?**

eslint flagged the `switch` with the message "Opening curly brace does not appear on the same line as controlling statement.".

**What did you expect to happen?**

Nothing. With "allman" style the opening curly brace should be on the next line.

I have a fix, I'll be submitting a PR.

**Comment by eslintbot** [comment]
Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by aparajita** [comment]
I looked around and I could not find an example of `switch` in the `allman` style, but since the `allman` style is all about putting braces separate from the control statement, it seems to me the current implementation is incorrect.

**Comment by ilyavolodin** [comment]
Reproducible in the online demo. I agree. In allman style all braces should be on separate lines.

**Comment by lo1tuma** [comment]
Looks like a duplicate of #3903.

**Comment by ilyavolodin** [comment]
Ahh! Thanks @lo1tuma Forgot that it's already been fixed. Closing in favor of #3903 which has already been fixed.

**Comment by aparajita** [comment]
Sorry guys, but that code is wrong. Comparing with "1tbs" is wrong, because "stroustrop" then fails to do the right thing. My version is correct.

**Comment by lo1tuma** [comment]
@aparajita So you are saying the fix for #3903 introduced a new bug? If so, can you please file a new issue an give an example what kind of code causes the new bug?

**Comment by aparajita** [comment]
Sorry, I don't have time. My version works.

**Comment by aparajita** [comment]
@lo1tuma It's quite unfair for you to ask me to show you why a wrong fix was accepted when I made the correct fix already. Isn't it your responsibility to verify the fixes?

**Comment by lo1tuma** [comment]
@aparajita I haven’t asked you why the fix for #3903 was accepted. My question was what kind of new bug this fix introduced. Anyway @ilyavolodin filed an issue for this #4004.
